File: pageobjects/src/test/java/AutoTestJavaStatics/PageObjects/WpPage.java
package autoTestJavaStatics.pageObjects;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

/**
 * Created by Kuba on 2017-05-30.
 */
public abstract class WpPage {

    public static final String MAIN_PAGE_URL = "http://autotestjava.wordpress.com";
    protected static final By FOOTER_LOCATOR = By.tagName("footer");

    protected static void WaitForElementPresent(By byLocator, WebDriver driver) {
        WebDriverWait wait = new WebDriverWait(driver, 20);
        wait.until(ExpectedConditions.elementToBeClickable(byLocator));
    }

    protected static void WaitUntilFooterIsDisplayed(WebDriver driver) {
        WaitForElementPresent(WpPage.FOOTER_LOCATOR, driver);
    }
}

File: VisualStudioPageObjects/VisualStudioPageObjects/autoTestJavaStatics/PageObjects/WpPage.cs
using OpenQA.Selenium;
using OpenQA.Selenium.Support.UI;
using System;

namespace autoTestJavaStatics.PageObjects
{
    abstract class WpPage
    {
        protected IWebDriver driver;
        public static readonly string URL_MAIN_PAGE = "http://autotestjava.wordpress.com";
        protected static readonly By LOCATOR_FOOTER = By.TagName("footer");

        protected static void WaitUntilElementIsClickable(By byLocator, IWebDriver driver)
        {
            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(20));
            wait.Until(ExpectedConditions.ElementToBeClickable(byLocator));
        }

        protected static void WaitUntilElementIsVisible(By byLocator, IWebDriver driver)
        {
            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(20));
            wait.Until(ExpectedConditions.ElementToBeClickable(byLocator));
        }

        protected static void WaitUntilElementIsHidden(By byLocator, IWebDriver driver)
        {
            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(20));
            wait.Until(ExpectedConditions.InvisibilityOfElementLocated(byLocator));
        }

        protected static void WaitUntilFooterIsDisplayed(IWebDriver driver)
        {
            WaitUntilElementIsVisible(WpPage.LOCATOR_FOOTER, driver);
        }
    }
}
